pages/spotlight/[slug]/compare.tsx

```diff
@@ -18,16 +20,32 @@ interface CompareProps {
 }
 
 const Compare: NextPage<CompareProps> = ({ setData, scaffold, page }) => {
+  const router = useRouter();
```

Comment:
If this isn't going to be relevant to this file, it should't be here. I suggest u move it to the component that needs it. You passed it too early and then had to keep passing the locations from one component to another just to get them to LocationComparisonBanner. Why not just do all this in there?